src/functions.php

```diff
@@ -54,8 +56,7 @@ function resolve($promiseOrValue): PromiseInterface
  * throwing an exception. For example, it allows you to propagate a rejection with
  * the value of another promise.
  *
- * @param \Throwable $reason
- * @return PromiseInterface
+ * @return PromiseInterface<null>
```

Comment:
Shouldn't this be `PromiseInterface<never>`?

Should probably add some tests similar to this:

```php
assertType('React\Promise\PromiseInterface<never>', reject(new \RuntimeException()));

assertType('React\Promise\PromiseInterface<never>', reject(new \RuntimeException())->then(fn(): int => 42));

assertType('React\Promise\PromiseInterface<int>', reject(new \RuntimeException())->then(null, fn(): int => 42));

assertType('React\Promise\PromiseInterface<int>', reject(new \RuntimeException())->catch(fn(): int => 42));
```

Not sure how much of this is supported, WDYT?

Reply:
Same issue as fullfilled promise type chaging during the chain:

```
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   tests/types/reject.php                                                                                                                                                                      
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  8      Expected type React\Promise\PromiseInterface<never>, actual: React\Promise\PromiseInterface<*NEVER*>                                                                                        
  9      Expected type React\Promise\PromiseInterface<never>, actual: React\Promise\PromiseInterface<*NEVER*>                                                                                        
  9      Parameter #1 $onFulfilled of method React\Promise\PromiseInterface<*NEVER*>::then() expects (callable(never): (React\Promise\PromiseInterface<never>|void))|null, 'React\\Promise\\Types…'  
         given.                                                                                                                                                                                      
  10     Expected type React\Promise\PromiseInterface<int>, actual: React\Promise\PromiseInterface<*NEVER*>                                                                                          
  11     Expected type React\Promise\PromiseInterface<int>, actual: React\Promise\PromiseInterface<*NEVER*>                                                                                          
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

```

Reply:
So the trick for line 8 and 9 is to literally use `*NEVER*` instead of `never` :facepalm: